lib/Exception/ServiceException.php

```diff
@@ -24,6 +24,18 @@
 namespace OCA\Mail\Exception;
 
 use Exception;
+use Throwable;
 
 class ServiceException extends Exception {
+	/**
+	 * @param string $message [optional] The Exception message to throw.
+	 * @param mixed $code [optional] The Exception code.
+	 * @param null|Throwable $previous [optional] The previous throwable used for the exception chaining.
+	 */
+	public function __construct($message = "", $code = 0, Throwable $previous = null) {
+		if (!is_int($code)) {
```

Comment:
@ChristophWurst I'm unsure about the constructor. Psalm mentioned the redundant int casts. ServiceException is often used as wrapper for Horde exceptions. I can't say for sure that every Horde exception is already using a numeric error code.

Reply:
I remember a time where it was not guaranteed that the code would be an int. This might have been fixed in PHP or the stubs. If psalm no longer complains without the cast then remove it.